### PR TITLE
Add date queries to the Customers endpoint

### PR DESCRIPTION
**Pull request by cklosowski**
#3157 #4602

### CODE REVIEW

File: includes/api/class-edd-api-v2.php [hunk 1]
```diff
@@ -148,4 +148,152 @@ public function get_products( $args = array() ) {
...
+			}
+
+			if ( ! empty( $args['enddate'] ) ) {
+				$date_range['end'] = $dates['year_end'] . sprintf('%02d', $dates['m_end'] )   . $dates['day_end'];
+			}
+
+			$args['date'] = $date_range;
+
+		} else {
+			if( $args['date'] == 'this_quarter' || $args['date'] == 'last_quarter'  ) {
```
Comment (pippinsplugins) [inline includes/api/class-edd-api-v2.php]: Can we add the other pre-defined date ranges too please? Today, yesterday, this week, last week.

Comment (cklosowski) [inline includes/api/class-edd-api-v2.php]: I need to add `*_week`, but `yesterday/today` are supported already. They fallback to the last case here on line 231

Comment (cklosowski) [inline includes/api/class-edd-api-v2.php]: @pippinsplugins I don't see `this_week` or `last_week` currently supported in the `get_dates` method of the existing API.

Comment (pippinsplugins) [inline includes/api/class-edd-api-v2.php]: Huh, they're in the reports UI but not in the REST API. Oops :P

**Pull request merged**